src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -249,17 +255,17 @@ public void SetTransactionMode(TransactionMode mode)
         /// <typeparam name="T"></typeparam>
         /// <param name="instance">The instance.</param>
         /// <returns></returns>
-        public List<string> GetCountersFor<T>(T instance)
+        public HashSet<string> GetCountersFor<T>(T instance)
         {
             if (instance == null)
                 throw new ArgumentNullException(nameof(instance));
 
             var documentInfo = GetDocumentInfo(instance);
 
-            if (documentInfo.Metadata.TryGet(Constants.Documents.Metadata.Counters, out BlittableJsonReaderArray counters))
-                return counters.Select(x=>x.ToString()).ToList();
-
-            return null;
+            if (documentInfo.Metadata.TryGet(Constants.Documents.Metadata.Counters, 
+                out BlittableJsonReaderArray counters) == false)
+                return null;
+            return new HashSet<string>(counters.Select(x=>x.ToString()), StringComparer.OrdinalIgnoreCase);
```

Comment:
Is there a scenario where we _can_ have duplicate counter names?
And are counter names really case insensitive?

Reply:
reverted back to List